### ISSUE TITLE
How select all likes by user order by added date?

### ISSUE THREAD

**Issue by magnum-ferox** [cybercog/laravel-love#13]
Hello. I need to show all likes by user in descending chronologycal order.
But I see in code only ScopeOrderByLikesCount..,
How make it?

**Comment by a-komarev** [comment]
Good question @magnum-ferox!

`scopeOrderByLikesCount` method if used to order `Likeables` by likes count within one model type. For example if you want to compare likes of all the `Articles`.

```php
$articlesOrderedByLikes = Articles::orderByLikesCount('desc')->get();
```

This feature is in [package to do list](https://github.com/cybercog/laravel-love/issues/7#issuecomment-357564132) and not implemented yet. Don't have enough time to continue refactoring of the API at this moment.

As a quick workaround you could add method to your `User` model:
```php
public function likes()
{
    return $this->hasMany(Like::class, 'user_id');
}
```

And then you will be able to get likes:
```php
$likes = $user->likes()->get();
```

If you want to have `Likeable` entities inside of each `Like` you could use method `with`:
```php
$likesWithLikeables = $user->likes()->with('likeable')->get();
```

To order collection by `Like` in chronological order just use default Eloquent Query Builder methods:
```php
$likesDescending = $user->likes()->with('likeable')->latest('id')->get();
$likesAscending = $user->likes()->with('likeable')->oldest('id')->get();
```

Hope this helps!

**Comment by antonkomarev** [comment]
This feature will be available after PR #24 will be merged.

API described below.

If you need to get all reactions which user has made:

```php
$reactions = $user->getReacter()
    ->reactions()
    ->latest('id')
    ->get();
```

If you need to get only reactions of `Like` type which user has made:

```php
$likeType = ReactionType::fromName('Like');
$reactions = $user->getReacter()
    ->reactions()
    ->where('reaction_type_id', $likeType->id)
    ->latest('id')
    ->get();
```

**Comment by antonkomarev** [comment]
PR #24 was merged. Laravel Love v6 will be released soon.

With [new API](https://github.com/cybercog/laravel-love/issues/13#issuecomment-428033102) we will be able to order reactions while collecting them.
